controllers/managedclustermodule_reconciler.go

```diff
@@ -82,25 +87,35 @@ func (r *ManagedClusterModuleReconciler) Reconcile(ctx context.Context, req ctrl
 
 	logger.Info("Requested KMMO ManagedClusterModule")
 
-	clusters, err := r.getSelectedManagedClusters(ctx, mcm)
+	clusters, err := r.clusterAPI.SelectedManagedClusters(ctx, mcm)
 	if err != nil {
 		return res, fmt.Errorf("failed to get selected clusters: %v", err)
 	}
 
 	for _, cluster := range clusters.Items {
+		logger := log.FromContext(ctx).WithValues("cluster", cluster.Name)
+		clusterCtx := log.IntoContext(ctx, logger)
+
+		if requeue := r.clusterAPI.BuildAndSign(clusterCtx, *mcm, cluster); requeue {
+			logger.Info("Build and Sign require a requeue; skipping ManifestWork reconciliation")
+			res.Requeue = true
+			continue
+		}
+
 		mw := &workv1.ManifestWork{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      mcm.Name,
 				Namespace: cluster.Name,
 			},
 		}
 
-		opRes, err := controllerutil.CreateOrPatch(ctx, r.client, mw, func() error {
+		opRes, err := controllerutil.CreateOrPatch(clusterCtx, r.client, mw, func() error {
 			return r.manifestAPI.SetManifestWorkAsDesired(ctx, mw, *mcm)
 		})
 
 		if err != nil {
-			return res, fmt.Errorf("failed to create/patch ManifestWork for managed cluster %s: %v", cluster.Name, err)
+			logger.Error(err, "failed to create/patch ManifestWork for managed cluster: %v", err)
+			continue
```

Comment:
Why do we skip other clusters here? Don't we want a best effort still trying to create a `ManifestWork` for other clusters?

Reply:
Yes, exactly. We don't skip other clusters, we are logging the error and then `continue`-ing the loop in order to try to reconcile the rest of the managed clusters, which the `ManagedClusterModule` selector is targeting.

Reply:
Seems like I forgot what `continue` is doing ;) 
I confused it with `break` but for loops haha.

Thanks. It makes perfect sense now.